ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ApprovedDeferrals.tsx

```diff
@@ -170,34 +164,91 @@ function ApprovedDeferrals() {
                     </Tr>
                 </Thead>
                 <Tbody>
-                    {data.map((exception) => {
-                        const { id, name, requester, createdAt, scope } = exception;
-                        return (
-                            <Tr key={id}>
-                                <Td>
-                                    <RequestIDLink id={id} name={name} context="CURRENT" />
-                                </Td>
-                                <Td>
-                                    <Requester requester={requester} />
-                                </Td>
-                                <Td>
-                                    <RequestedAction exception={exception} context="CURRENT" />
-                                </Td>
-                                <Td>
-                                    <RequestCreatedAt createdAt={createdAt} />
-                                </Td>
-                                <Td>
-                                    <RequestExpires exception={exception} context="CURRENT" />
-                                </Td>
-                                <Td>
-                                    <RequestScope scope={scope} />
-                                </Td>
-                                <Td>
-                                    <RequestedItems exception={exception} context="CURRENT" />
-                                </Td>
-                            </Tr>
-                        );
-                    })}
+                    {tableUIState.type === 'LOADING' && (
```

Comment:
👍 This mutually exclusive way of rendering based on state is much cleaner to read compared to the early returns.

Reply:
Maybe because I worked in a subject area which had as common idiom a sequence of conditions from special cases to general cases, mutually exclusive cases cause me to stop and analyze possible states to make sure the component never renders two inconsistent states.

Also, compared to a plural entities page like this for which default `[]` is renderable, without careful design of states, singular entity page for which `null` is pre-fetch state, component might need to guard `COMPLETE` state with a `!== null` condition for TypeScript. Before we finalize the pattern, we need an example to critique.

Reply:
Mark has some good points. The rendering logic will be very similar across different use cases, so we might consider creating a reusable component for the **Tbody** to do the rendering logic within that component. The **type** should be very distinct so we shouldn't have weird states. If we create a proper component for the rendering logic within the **Tbody** we should also be confident there won't be any weird states. I agree we should try this out with an example to critique. Maybe something that does polling so we can have a complete critique.

Reply:
Also, on the point of the "singular entity page", I was thinking we could consider a separate function to deal with that. I would prefer not to make a function that is so general that we use it for all cases. It would make it a bit difficult to maintain and refactor as we see changes in UX long-term.

Reply:
Also, I think I should have made a test for that function. I'll add that as well